### PR TITLE
Add cmake script + tests + github actions support

### PR DESCRIPTION
**Pull request by madebr**
Hello,

this pr improves the cmake script by:
- allowing static + shared (including allowing dll's on windows)
- adds installation though cmake
- adds testing using ctest: it compares the output of the test and the java executable
- adds CI by making use of github actions

The tests currently fail due to a timeout and assertion error.
See https://github.com/madebr/libproperties/actions/runs/419190631 for a test run.

### CODE REVIEW

File: properties.h [hunk 1]
```diff
@@ -1,13 +1,25 @@
...
 #else
-#define EXTERN
+#define LIBPROPERTIES_EXTERN
+#endif
+#else
+#define LIBPROPERTIES_EXTERN
+#endif
+
+#ifdef __cplusplus
+#define EXTERN extern "C" {
```
Comment (madebr) [inline properties.h:21]:
```suggestion
extern "C" {
```

File: CMakeLists.txt [hunk 2]
```diff
@@ -10,9 +10,6 @@ project(libproperties VERSION ${MY_VERSION} LANGUAGES C)
 #   Define GNU standard installation directories
 include(GNUInstallDirs)
 
-#   We need -fsigned-char for ARM compiler
-set(CMAKE_C_FLAGS "-fsigned-char ${CMAKE_C_FLAGS}")
```
Comment (libbylg) [inline CMakeLists.txt:14]: We need the `-fsigned-char` on ARM, so we should not remove it.

File: properties.h [hunk 3]
```diff
@@ -1,6 +1,22 @@
...
+#define LIBPROPERTIES_EXTERN __attribute__((visibility("default")))
+#endif
+#elif !defined(LIBPROPERTIES_STATIC)
+#ifdef _WIN32
+#define LIBPROPERTIES_EXTERN __declspec(dllimport)
+#else
+#define LIBPROPERTIES_EXTERN
+#endif
+#else
+#define LIBPROPERTIES_EXTERN
```
Comment (libbylg) [inline properties.h]: We'd better use the name `LIBPROPERTIES_API`.

Comment (madebr) [reply properties.h]: I will make the change at the next opportunity

Comment (libbylg) [reply properties.h]: I have solved.

File: properties.h (hunk 1 above)
Comment (madebr) [reply properties.h:21]: @libbylg
Don't forget to fix this one.
Maybe also add a test, using your library by a c++ source

**Pull request merged**